### PR TITLE
Rise Bid Adapter: improve isBidRequestValid and size detection along with other updates

### PR DESCRIPTION
**Pull request by noamtzu**
## Type of change

- [X] Bugfix
- [x] Feature
- [X] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)

## Description of change
1. Add global vendor ID
2. Improve isBidRequestValid function
3. Improve size detection
4. Fix floor price bug in case of pbjs version 5 + floor price in bidder param
5. Add support for: dnt,deviceType,playbackMethod,placement,pos,minduration,maxduration,skip,linearity
6. Support all user sync modules
7. Support first party data

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@noamtzu can you please remove the package-lock file from this pr 🙏 Also I see there are no new tests or update to testing?

**Comment by noamtzu** [comment]
Hi @ChrisHuie , Done. 
No changes needs in our tests. Thanks 🙏

File: modules/riseBidAdapter.js [hunk 1]
```diff
@@ -126,12 +137,22 @@ function buildVideoRequest(bid, bidderRequest) {
...
 function getSizes(bid) {
-  if (utils.deepAccess(bid, 'mediaTypes.video.sizes')) {
-    return bid.mediaTypes.video.sizes[0];
-  } else if (Array.isArray(bid.sizes) && bid.sizes.length > 0) {
-    return bid.sizes[0];
+  let playerSize = utils.deepAccess(bid, 'mediaTypes.video.playerSize');
+  let width;
+  let height;
+
+  if (utils.isArray(bid.sizes) && bid.sizes.length === 2 && !utils.isArray(bid.sizes[0])) {
```
Comment (Fawke) [inline modules/riseBidAdapter.js:144]: Adding this is fine but FYI, the ad unit is already checked for valid sizes and it's made sure that it's in the right format. Check [this](https://github.com/prebid/Prebid.js/blob/697abe06a4b63bc64635688a69beda084d2dd7e6/src/prebid.js#L75) piece of code.

**Review by Fawke — CHANGES_REQUESTED** [review]
Hi @noamtzu,

We'd need a docs PR for a GVLID addition, it'll go under `gvl_id`.

File: modules/riseBidAdapter.js (hunk 1 above)
Comment (noamtzu) [reply modules/riseBidAdapter.js]: Removed

**Comment by noamtzu** [comment]
@Fawke 
Thank you for your note.
Docs: https://github.com/prebid/prebid.github.io/pull/3248

**Comment by noamtzu** [comment]
@Fawke My circle has not passed due to another adapter. is it possible?
https://app.circleci.com/pipelines/github/prebid/Prebid.js/7937/workflows/f08e7982-529c-4ffb-8190-2a14c26a1bf2/jobs/17770

**Comment by Fawke** [comment]
@noamtzu Sometimes the tests of some module fails, I checked it wasn't related to your submission. I rerun it to make it pass.

**Review by Fawke — APPROVED** [review]

**Pull request merged**